tt_metal/llrt/tt_cluster.cpp

```diff
@@ -179,6 +179,9 @@ void Cluster::initialize_device_drivers() {
 
     tt_device_params default_params;
     this->start_driver(default_params);
+    this->generate_virtual_to_umd_coord_mapping();
```

Comment:
is this called virtual to umd coord mapping to allow umd to change coord without Metal having to uplift everything. currently umd accepts virtual coordinates but this could change? if I understand that correctly could you just add a comment explaining that somewhere

Reply:
UMD defines Virtual Coordinates differently from how TT-Metal defines them. In UMD, Virtual Coordinates are in the Physical space, but the Tensix cores are packed together (i.e. harvested rows are moved to the bottom of the grid). See [here](https://yyz-gitlab.local.tenstorrent.com/tenstorrent/open-umd/-/wikis/home/Open-UMD-Coordinates-Systems#virtualized-coordinates-inputs-and-outputs-from-umd) for UMD's definition of Virtual Coordinates.

In TT-Metal, Virtual Coordinates are in the Translated Space. The tables being generated here allow converting coordinates in the Translated Space to UMD Virtual Coordinates. Eventually, the goal is to allow UMD to accept all coordinate types, at which point this mapping would be unnecessary. We would simply pass in the coordinates we use across the Metal stack to UMD.

Also, IIRC we also plan on removing the current UMD Virtual Coordinate scheme entirely.

This mapping is here to make uplifting easier once the UMD Coordinate Assignment scheme is mainlined. I'll add a comment for this.

Reply:
nice thanks. also kinda related question, do we want to keep using noc0 coordinates in the soc desc yamls

Reply:
NOC0 Coordinates in the SOC Descs are useful for things like deriving the optimal physical placement of worker cores and ensuring that debug tools such as NOC sanitize still work when NOC HW reprograms registers with physical coords.
We could add the virtual SOC descriptor files to the repo, but I don't see the point atm. Its fairly easy to use cluster APIs to derive virtual coords from NOC0 coords.